### ISSUE TITLE
how to generate the training data in 4D TIFF format

### ISSUE THREAD

**Issue by jinmeng** [mahmoudnafifi/Semantic-Color-Constancy-Using-CNN#1]
HI, I am a Newbie, I would like to retrain SCCNet, would you please tell me how to generate the training data in 4D TIFF format? 

Thanks very much

**Comment by mahmoudnafifi** [comment]
Hi,

You can use the ground-truth semantic masks in [ADE20K dataset](https://groups.csail.mit.edu/vision/datasets/ADE20K/) and append the semantic mask as a fourth channel to your three-channel image. Also, remember that this paper is a proof of concept of the idea of using the semantic mask for color constancy. So we use the 8-bit sRGB images not raw images as there are no semantic masks available for raw images. But in reality, color constancy always applied to the linear raw-RGB images and applying the diagonal correction on rendered sRGB images cannot always give good results. See this [paper](http://openaccess.thecvf.com/content_CVPR_2019/papers/Afifi_When_Color_Constancy_Goes_Wrong_Correcting_Improperly_White-Balanced_Images_CVPR_2019_paper.pdf) for more information.

Hope this helps.

**Comment by jinmeng** [comment]
Thanks very much. I read you matlab code and now know how to use it

Best Regards

Mahmoud Afifi <notifications@github.com> 于2020年3月16日周一 上午11:54写道：

**Comment by mahmoudnafifi** [comment]
Happy to hear that.
